### PR TITLE
HDFS-9895. Push up DataNode#conf to base class

### PR DESCRIPTION
**Pull request by xiaobingo**
Please kindly review the patch v001, see also https://issues.apache.org/jira/browse/HDFS-9895.

### CODE REVIEW

File: hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/datanode/DNConf.java [hunk 1]
```diff
@@ -113,71 +112,71 @@
...
   private final boolean allowNonLocalLazyPersist;
-
+  private final DataNode dn;
   private final int volFailuresTolerated;
   private final int volsConfigured;
 
-  public DNConf(Configuration conf) {
-    this.conf = conf;
-    socketTimeout = conf.getInt(DFS_CLIENT_SOCKET_TIMEOUT_KEY,
+  public DNConf(final DataNode dn) {
```
Comment (arp7) [inline hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/datanode/DNConf.java]: The dn.getConf() object is not referenced outside the constructor so you can just pass a reference to that object. Also DNConf need not keep a reference to the dn. I think you can just revert all changes to this file.

File: hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/datanode/DataNode.java [hunk 2]
```diff
@@ -1239,9 +1236,8 @@ void startDataNode(Configuration conf,
     synchronized (this) {
       this.dataDirs = dataDirs;
     }
-    this.conf = conf;
-    this.dnConf = new DNConf(conf);
-    checkSecureConfig(dnConf, conf, resources);
+    this.dnConf = new DNConf(this);
```
Comment (arp7) [inline hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/datanode/DataNode.java]: If we revert changes to DNConf we can just replace this with `this.dnConf = new DNConf(getConf())`.

**Comment by elek** [comment]
Can one of the admins verify this patch?

**Comment by aajisaka** [comment]
https://issues.apache.org/jira/browse/HDFS-9895 has been fixed. Closing.

**Pull request closed without merging**